tendermint/src/lite/verifier.rs

```diff
@@ -403,24 +403,24 @@ mod tests {
     // Init a mock Requester.
     // For each pair of lists of validators (cur and next vals) we
     // init a trusted state (signed header and vals);
+    // Note: from the final_state we only feed the validators into the Requester.
     // these are (signed) headers and validators will be returned by the Requester.
-    fn init_requester(vals_for_height: Vec<Vec<usize>>) -> MockRequester {
+    fn init_requester(vals_for_height: Vec<Vec<usize>>, final_state: &MockState) -> MockRequester {
```

Comment:
I think I meant make vals_for_height one height longer here, which would make this fn simpler, and would also decouple the expected final_state from the last thing the requester reports, since they can be different if the requester is faulty.

Reply:
The way we currently init the requester would mean, the vector vals_for_height needs to be 2 heights longer (e.g. if we want to bisect to height 5, we need vals up-to 6, and to write those into the requester, we need the next_vals for them, i.e. 7 here). Does that make sense?